wcfsetup/install/files/lib/system/form/builder/field/FileProcessorFormField.class.php

```diff
@@ -203,6 +203,18 @@ public function validate()
             );
         }
 
+        foreach ($this->files as $file) {
+            if (!FileProcessor::getInstance()->canAdopt($fileProcessor, $file, $this->context)) {
+                $this->addValidationError(
+                    new FormFieldValidationError(
+                        'adopt',
+                        'wcf.form.field.fileProcessor.error.adopt',
```

Comment:
Should this be a more generic error message because it could happen outside of the form builder too?